plugins/woocommerce/includes/class-wc-form-handler.php

```diff
@@ -929,10 +929,17 @@ private static function add_to_cart_handler_variable( $product_id ) {
 	 * @throws Exception On login error.
 	 */
 	public static function process_login() {
+
+		static $valid_nonce;
+
 		// The global form-login.php template used `_wpnonce` in template versions < 3.3.0.
 		$nonce_value = wc_get_var( $_REQUEST['woocommerce-login-nonce'], wc_get_var( $_REQUEST['_wpnonce'], '' ) ); // @codingStandardsIgnoreLine.
 
-		if ( isset( $_POST['login'], $_POST['username'], $_POST['password'] ) && wp_verify_nonce( $nonce_value, 'woocommerce-login' ) ) {
+		if ( ! empty( $nonce_value ) ) {
```

Comment:
If you moved this to above line 936 you could change it to

```suggestion
		if ( null === $valid_nonce ) {
```

This would ensure the nonce logic only ran once.